File: examples/catFactFetcher/output.cs
using System;
using System.Net.Http;
using System.Threading.Tasks;
using Newtonsoft.Json.Linq;

public class Program
{
    private static readonly HttpClient client = new HttpClient();

    public static async Task Main(string[] args)
    {
        try
        {
            // Send GET request
            Console.WriteLine("Sending GET request...");
            var response = await client.GetAsync("https://meowfacts.herokuapp.com/?count=1");

            // Make sure the http response is 200
            if (response.IsSuccessStatusCode)
            {
                Console.WriteLine("Response received...");

                // Decode JSON response
                Console.WriteLine("Decoding JSON response...");
                var jsonString = await response.Content.ReadAsStringAsync();
                var jsonObject = JObject.Parse(jsonString);

                // Read Fact
                Console.WriteLine("Reading Fact...");
                var fact = jsonObject["data"][0].ToString();

                // Return Fact
                Console.WriteLine("Returning Fact...");
                Console.WriteLine(fact);
            }
            else
            {
                Console.WriteLine("HTTP Response is not 200. Exiting program...");
            }
        }
        catch (Exception e)
        {
            // Stop program if any error happened
            Console.WriteLine("An error occurred: " + e.Message);
        }
    }
}

File: examples/catFactFetcher/output.py
import requests
import json

def GetRandomFact():
    # Send GET request
    print("Sending GET request...")
    response = requests.get("https://meowfacts.herokuapp.com/?count=1")

    # Make sure the http response is 200 and the response is valid
    if response.status_code != 200:
        print("Error occurred: HTTP response is not 200.")
        return None

    # Decode JSON response
    print("Decoding JSON response...")
    try:
        json_response = response.json()
    except json.JSONDecodeError:
        print("Error occurred: Unable to decode JSON response.")
        return None

    # Read Fact
    print("Reading fact...")
    try:
        fact = json_response['data'][0]
    except (KeyError, IndexError):
        print("Error occurred: Unable to read fact from JSON response.")
        return None

    # Return Fact
    print("Returning fact...")
    return fact

# Call the function
print(GetRandomFact())
